flowmachine/flowmachine/features/spatial/distance_matrix.py

```diff
@@ -49,21 +77,41 @@ def __init__(
             self.spatial_unit = spatial_unit
 
         self.spatial_unit.verify_criterion("has_lon_lat_columns")
-
+        self.geom_matrix = _GeomDistanceMatrix(geom_table=self.spatial_unit.geom_table)
         self.return_geometry = return_geometry
 
         super().__init__()
 
+    @property
+    def table_name(self):
+        raise NotImplementedError("Store this object's geom_matrix instead.")
+
     @property
     def column_names(self) -> List[str]:
         col_names = [f"{c}_from" for c in self.spatial_unit.location_id_columns]
         col_names += [f"{c}_to" for c in self.spatial_unit.location_id_columns]
-        col_names += ["distance"]
+        col_names += ["value"]
         if self.return_geometry:
             col_names += ["geom_origin", "geom_destination"]
         return col_names
 
     def _make_query(self):
+        if self.spatial_unit == LonLatSpatialUnit():
```

Comment:
I don't like to re-introduce an if/else for handling different spatial units, but I don't think there's an easy way around it in this case.